integrations/docker/images/stage-2/chip-build-linux-qemu/Dockerfile

```diff
@@ -0,0 +1,216 @@
+	sh 'ssh-keygen -A' : \
+	sh '/bin/echo -e "PermitRootLogin yes\nPasswordAuthentication yes\nPermitEmptyPasswords yes" > /etc/ssh/sshd_config' : \
+	mkdir-p "/etc/netplan" : \
+	sh '/bin/echo -e "network:\n  version: 2\n  renderer: networkd\n  ethernets:\n    enp0s4:\n      dhcp4: true\n" > /etc/netplan/01-netcfg.yaml' : \
+	sh 'chmod -R 700 /etc/netplan' : \
+	sh 'sed -i "s#^ExecStart=.*#ExecStart=-/sbin/agetty -o \"-p -- \\\\\\\\u\" -a root --keep-baud 115200,38400,9600 %I \$TERM#" "/usr/lib/systemd/system/serial-getty@.service"' : \
+	mkdir-p "/etc/bluetooth" : \
+	copy-in /tmp/workdir/main.conf /etc/bluetooth : \
+	sh 'sed -i "s#^ExecStart=.*#ExecStart=-/usr/libexec/bluetooth/bluetoothd -E#" /lib/systemd/system/bluetooth.service' : \
+	sh 'rm -f /etc/resolv.conf && /bin/echo -e "nameserver 8.8.8.8" > /etc/resolv.conf' : \
+	sh '/bin/echo -e "host0 /chip 9p trans=virtio,version=9p2000.L 0 0" >> /etc/fstab' : \
+	sh '/bin/echo -e "export PW_ENVIRONMENT_ROOT=/root/pw_root\n[ -x /launcher.sh ] && /launcher.sh\n" >> /root/.profile' : \
+	sh 'DEBIAN_FRONTEND=noninteractive apt-get -y install git gcc g++ pkg-config libssl-dev libdbus-1-dev libglib2.0-dev libavahi-client-dev ninja-build python3 python3-venv python3-dev python3-pip unzip libgirepository1.0-dev libcairo2-dev libreadline-dev' : \
+	sh 'git config --file /root/.gitconfig  --add safe.directory "*"' : \
+	sh 'apt-get clean' : \
+	sh 'rm -rf /var/lib/apt/lists/*' : \
+	sh 'rm -rf /var/cache/apt/*' : \
+	sh 'echo Configuration completed.' \
+	&& mkdir -p /chip \
+	&& rm -rf /opt/ubuntu-qemu/rootfs \
+	&& echo -n \
+	"#!/bin/bash\n" \
+	"grep -q 'rootshell' /proc/cmdline && exit\n" \
+	"if [[ -x /chip/runner.sh ]]; then\n" \
+	"  echo '### RUNNER START ###'\n" \
+	"  cd /chip\n" \
+	"  bash /chip/runner.sh\n" \
+	"  status=\$?\n" \
+	"  echo \"### RUNNER STOP, RETURN: \$status\"\n" \
+	"  echo \$status > /chip/runner_status\n" \
+	"else\n" \
+	"  read -r -t 5 -p 'Press ENTER to access root shell...' && exit || echo ' timeout.'\n" \
+	"fi\n" \
+	"echo 'Shutting down emulated system...'\n" \
+	"echo o > /proc/sysrq-trigger\n" \
+	| guestfish --rw -a $UBUNTU_QEMU_IMG -m /dev/sda3:/ upload - /launcher.sh : chmod 0755 /launcher.sh \
+	&& virt-sparsify --compress ${UBUNTU_QEMU_IMG} ${UBUNTU_QEMU_IMG}.compressed \
+	&& mv ${UBUNTU_QEMU_IMG}.compressed ${UBUNTU_QEMU_IMG} \
+	&& rm -rf /var/tmp/.guestfs-0/* \
+	&& rm -rf /tmp/* \
+	&& : # last line
+
+FROM ghcr.io/project-chip/chip-build:${VERSION}
+
+ARG UBUNTU_QEMU_DIR_DEFAULT
+ARG UBUNTU_QEMU_IMG_DEFAULT
+
+ENV UBUNTU_QEMU_DIR=${UBUNTU_QEMU_DIR_DEFAULT}
+ENV UBUNTU_QEMU_IMG=${UBUNTU_QEMU_IMG_DEFAULT}
+ENV PW_ENVIRONMENT_ROOT="/root/pw_root"
+
+RUN set -x \
+	&& apt-get update \
+	&& DEBIAN_FRONTEND=noninteractive apt-get install -fy \
+	cpu-checker \
+	qemu \
+	qemu-system-x86 \
+	&& apt-get clean \
+	&& rm -rf /var/lib/apt/lists/* \
+	&& rm -rf /var/cache/apt/* \
+	&& : # last line
+COPY --from=build-env ${UBUNTU_QEMU_DIR} ${UBUNTU_QEMU_DIR}
```

Comment:
We try to place every possible build and sdk into `chip-build-vscode`. Should this be done for the qemu image as well or is it too large or not viable to run in the chip-build-vscode image?

Reply:
The size of layers added to ghcr.io/project-chip/chip-build is about 400Mb. I feel like we can add it to chip-build-vscode. I just didn't think about it earlier.